fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/RecursiveUnionExpression.java

```diff
@@ -56,17 +56,52 @@ public class RecursiveUnionExpression implements RelationalExpressionWithChildre
     @Nonnull
     private final CorrelationIdentifier tempTableInsertAlias;
 
+    @Nonnull
+    private final Traversal traversal;
+
     @Nonnull
     private final Value resultValue;
 
+    /**
+     * Defines the traversal strategy for recursive union operations in Common Table Expressions (CTEs).
+     * This enum specifies how the recursive leg of a recursive union should traverse and process
+     * the intermediate results during query execution.
+     */
+    public enum Traversal {
```

Comment:
Can  you name this something else? There already is a `Traversal` class in the planner which may lead to confusion.

Reply:
`StrategyHint` as this seems to be a hint given from SQL?